drivers/eeprom/eeprom_mchp_xec.c

```diff
@@ -62,10 +272,20 @@ static int eeprom_xec_write(const struct device *dev, off_t offset,
 	k_mutex_lock(&lock, K_FOREVER);
 
 	/* EEPROM HW WRITE */
+	for (chunk_idx = 0; chunk_idx < len; chunk_idx += XEC_EEPROM_TRANSFER_SIZE_WRITE) {
+		if ((len-chunk_idx) < XEC_EEPROM_TRANSFER_SIZE_WRITE) {
+			chunk_size = (len-chunk_idx);
+			eeprom_xec_data_write_32_bytes(regs, &data_buf[chunk_idx],
+							chunk_size, (offset+chunk_idx));
+			break;
```

Comment:
Seems redundant. 
If the chunk_size updated for last packet (<32 bytes), that's good enough, cause outside if loop - same function `eeprom_xec_data_write_32_bytes` call executed with updated `chunk_size`.

Reply:
yes, i can try removing it.

Reply:
updated